### PR TITLE
String function bug fix

### PR DESCRIPTION
**Pull request by aziz-mu**
This commit fixes the bugs in issue https://github.com/kuzudb/kuzu/issues/938 as well as adding tests for each bug.

Note: The `left` command had some other behaviour specified for negative index. I changed the behaviour/test for now, but feel free to let me know which one is better.

### CODE REVIEW

File: src/function/list/operations/include/list_extract_operation.h [hunk 1]
```diff
@@ -36,8 +36,12 @@ struct ListExtract {
     }
 
     static inline void operation(ku_string_t& str, int64_t& idx, ku_string_t& result) {
-        auto pos = idx > 0 ? min(idx, (int64_t)str.len) : max(str.len + idx, (int64_t)0) + 1;
-        result.set((char*)(str.getData() + pos - 1), 1 /* length */);
+        if (str.len < idx || idx < 1) {
```
Comment (acquamarin) [inline src/function/list/operations/include/list_extract_operation.h:39]: Actually idx can be a negative number. eg. "123"[-2] which means the second last character in string "123". We should support this feature as duckdb also supports this.

**Review by acquamarin — CHANGES_REQUESTED** [review]

File: src/function/list/operations/include/list_extract_operation.h (hunk 1 above)
Comment (aziz-mu) [reply src/function/list/operations/include/list_extract_operation.h]: Done

Comment (aziz-mu) [reply src/function/list/operations/include/list_extract_operation.h]: Added test for this as well

File: test/test_files/tinySNB/function/string.test [hunk 2]
```diff
@@ -343,6 +343,18 @@ Huber
 #
 #
 
+-NAME LpadOutOfRange
+-QUERY MATCH (p:person) RETURN lpad(p.fName, -10, "t")
```
Comment (acquamarin) [inline test/test_files/tinySNB/function/string.test:347]: Can you add a test for rpad as well?

File: src/function/string/operations/include/left_operation.h [hunk 3]
```diff
@@ -18,8 +18,7 @@ struct Left {
 public:
     static inline void operation(
         ku_string_t& left, int64_t& right, ku_string_t& result, ValueVector& resultValueVector) {
-        auto len = right > 0 ? min(left.len, (uint32_t)right) :
-                               max(left.len + (uint32_t)right, (uint32_t)0u);
+        auto len = right > 0 ? min(left.len, (uint32_t)right) : 0u;
```
Comment (acquamarin) [inline src/function/string/operations/include/left_operation.h:21]: I think left/right operation can take negtaive index.
eg. 
```
D select left('123', -1) from test;
┌─────────────────┐
│ left('123', -1) │
├─────────────────┤
│ 12              │
└─────────────────┘
```
Also change code for right as well

File: test/test_files/tinySNB/function/string.test [hunk 4]
```diff
@@ -426,14 +438,14 @@ DEsWor
 -NAME LeftNegativeIdxStructuredStrAndLiteralInt
 -QUERY MATCH (p:person) RETURN left(p.fName, -3)
 ---- 8
```
Comment (acquamarin) [inline test/test_files/tinySNB/function/string.test:440]: Can you check the correctness of right with negative idx as well?

**Review by acquamarin — APPROVED** [review]

**Pull request merged**